crates/issue/src/tests.rs

```diff
@@ -269,15 +270,76 @@ fn test_cancel_issue_not_found_fails() {
 }
 
 #[test]
-fn test_cancel_issue_not_expired_fails() {
+fn test_cancel_issue_not_expired_and_not_requester_fails() {
     run_test(|| {
         ext::vault_registry::get_active_vault_from_id::<Test>
             .mock_safe(|_| MockResult::Return(Ok(init_zero_vault(VAULT))));
 
         let issue_id = request_issue_ok(USER, 3, VAULT);
         // issue period is 10, we issued at block 1, so at block 5 the cancel should fail
         <security::Pallet<Test>>::set_active_block_number(5);
-        assert_noop!(cancel_issue(USER, &issue_id), TestError::TimeNotExpired,);
+        assert_noop!(cancel_issue(3, &issue_id), TestError::TimeNotExpired);
+    })
+}
+
+#[test]
+fn test_cancel_issue_not_expired_and_requester_succeeds() {
+    run_test(|| {
+        ext::vault_registry::get_active_vault_from_id::<Test>
+            .mock_safe(|_| MockResult::Return(Ok(init_zero_vault(VAULT))));
+        ext::vault_registry::decrease_to_be_issued_tokens::<Test>.mock_safe(move |_, _| MockResult::Return(Ok(())));
+        ext::vault_registry::is_vault_liquidated::<Test>.mock_safe(move |_| MockResult::Return(Ok(false)));
+        ext::fee::get_issue_griefing_collateral::<Test>.mock_safe(move |_| MockResult::Return(Ok(griefing(100))));
+
+        let issue_id = request_issue_ok(USER, 300, VAULT);
+
+        unsafe {
+            // issue period is 10, we issued at block 1, so at block 4 the requester gets 70% griefing back
+            <security::Pallet<Test>>::set_active_block_number(4);
+
+            ext::vault_registry::transfer_funds::<Test>.mock_raw(|_, _, amount| {
+                assert_eq!(amount, &griefing(30));
```

Comment:
This is not a complete test - we don't check if `transfer_funds` actually gets called. Also we should check that the other 70 gets unlocked